### DISCUSSION TITLE
Support GCP ID Token authentication

### DISCUSSION THREAD

**Discussion by jlarrow** [grafana/grafana-infinity-datasource#584] [Ideas]
Currently, the auth configurations for Google JWT does not seem to support generating ID tokens.  Access tokens are not able to be used with Cloud Functions or Cloud Run — rather an ID token is necessary.

It would be helpful if this plugin supported Application Default Credentials or Explicitly configured with a Service Account key file.

This would allow users to have data sources from Cloud Functions / Cloud Run.

**Comment by yesoreyeram** [comment]
Can you add relevant documents to the APIs you are referring?  Also any examples of applicable.

**Reply by JordanDeBeer** [reply]
These wouldn't be Google APIs but rather APIs that developers create themselves that are hosted on Cloud Run or Cloud Functions.

These services require identity tokens, which can be generated from Service Account keys or via ADC credentials like this:
https://pkg.go.dev/google.golang.org/api/idtoken#example-NewTokenSource-SetAuthorizationHeader
or this
https://cloud.google.com/run/docs/authenticating/service-to-service#use_the_authentication_libraries

**Reply by yesoreyeram** [reply]
Seems not a hard one. But I don't have any test environment.. how hard it is create a test environment? Do you have any sample scripts/steps to create such test environment where I can test this..

**Reply by jlarrow** [reply]
Hi @yesoreyeram, sorry it has taken so long to respond — I was trying to find a way to provide resources for a test env.  Unfortunately, I'm unable to do so.  GCP does offer a free tier though.

I currently deploy with `make` which has many requisites.  I can assemble a script to deploy a function that requires authentication if that would help?

Thanks!

**Reply by yesoreyeram** [reply]
yes. any sample script may help.

**Reply by jlarrow** [reply]
@yesoreyeram, I have a script to create a function that requires auth.  What is the best way you get it to you?
Thanks

**Reply by yesoreyeram** [reply]
you can share any way you want.. preferably gist.github.com

**Reply by jlarrow** [reply]
https://gist.github.com/jlarrow/6456b3c394429fd95ead9ddae99f1823

**Reply by yesoreyeram** [reply]
Hiya.. Sorry for the delay.. I added to my backlog. Will let you know how this goes.

**Reply by jlarrow** [reply]
Hey Sriram, thanks this feature will be greatly appreciated!

**Reply by jlarrow** [reply]
Hello Sriram, just checking in to see if we're any closer to using an identity token?
